pinc/project_states.inc

```diff
@@ -43,247 +49,245 @@ function declare_project_state(
     $phase,
     $star_metal
 ) {
-    global $PROJECT_STATES_IN_ORDER;
-    global $project_state_medium_label_;
-    global $project_state_long_label_;
-    global $project_state_forum_;
-    global $project_state_phase_;
+    global $project_states;
     global $project_states_for_star_metal_;
 
     define($constant_name, $constant_value);
-    $PROJECT_STATES_IN_ORDER[] = $constant_value;
-    $project_state_medium_label_[$constant_value] = $medium_label;
-    $project_state_long_label_[$constant_value] = $long_label;
-    $project_state_forum_[$constant_value] = $forum;
-    $project_state_phase_[$constant_value] = $phase;
+
+    $project_states[$constant_value] = new ProjectState(
```

Comment:
constant_name is the string defining the constant so it resolves to the same thing as constant_value. constant_value is used as the key for the array of objects so there doesn't seem to be a use for the same thing inside the object.

We could put star_metal in the object and postpone constructing the sql condition until we need it. I'll try doing that.

Reply:
> We could put star_metal in the object and postpone constructing the sql condition until we need it. I'll try doing that.

This would be possible but would involve changing some other files so for now just move the sql constructor into a function that uses the objects.